submarine-server/server-core/src/main/java/org/apache/submarine/server/rest/JobManagerRestApi.java

```diff
@@ -59,24 +61,24 @@ public Response ping() {
 
   /**
    * Returns the contents of {@link Job} that submitted by user.
-   * @param jobSpec job spec
+   * @param spec job spec
    * @return the contents of job
    */
   @POST
   @Consumes({RestConstants.MEDIA_TYPE_YAML, MediaType.APPLICATION_JSON})
-  public Response submitJob(JobSpec jobSpec) {
-    if (!jobSpec.validate()) {
-      return new JsonResponse.Builder<String>(Response.Status.ACCEPTED)
-          .success(false).result("Invalid params.").build();
-    }
-
+  public Response createJob(JobSpec spec) {
     try {
-      Job job = JobManager.getInstance().submitJob(jobSpec);
-      return new JsonResponse.Builder<Job>(Response.Status.OK)
-          .success(true).result(job).build();
-    } catch (UnsupportedJobTypeException e) {
-      return new JsonResponse.Builder<String>(Response.Status.ACCEPTED)
-          .success(false).result(e.getMessage()).build();
+      Job job = JobManager.getInstance().createJob(spec);
```

Comment:
We should check if the job exists and return the proper response. This is important. Otherwise, a same k8s job spec with the same job name could be submitted but fail quietly without any hint to the user.